iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_ptr.inl

```diff
@@ -83,23 +84,24 @@ inline T* unique_ptr<T>::get() noexcept
 template <typename T>
 inline const T* unique_ptr<T>::get() const noexcept
 {
-    // AXIVION Next Construct AutosarC++19_03-A5.2.3 : Avoid code duplication
+    // AXIVION Next Construct AutosarC++19_03-A5.2.3, CertC++-EXP55 : Avoid code duplication
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
     return const_cast<unique_ptr<T>*>(this)->get();
 }
 
 template <typename T>
 inline T* unique_ptr<T>::release(unique_ptr&& ptrToBeReleased) noexcept
 {
-    auto ptr = ptrToBeReleased.m_ptr;
+    // AXIVION Next Line AutosarC++19_03-A7.1.1 : Pointer is explicitly not const to adhere to return type
```

Comment:
No this warning is correct. Just adjust the code to `auto* const ptr`.

We want here a constant pointer which point to a mutable T. I think you mixed it up with a mutable pointer which points to a constant T.